google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/BlobWriteChannel.java

```diff
@@ -83,6 +92,26 @@ public String call() {
     }
   }
 
+  private static String open(final URL signURL, final StorageOptions options) {
+    try {
+      return runWithRetries(
+          new Callable<String>() {
+            @Override
+            public String call() {
+              if (!signURL.getQuery().contains("&Signature=")) {
```

Comment:
Is this a sufficient check for signature validity? Why not check for other fields as well?